File: backend/app/schemas/inventario.py
from datetime import datetime
from decimal import Decimal
from enum import Enum

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import UnidadInventario


class TipoAjuste(str, Enum):
    entrada = "entrada"
    salida = "salida"


class InventarioOut(BaseModel):
    id: int
    id_producto: int
    cantidad: Decimal
    stock_minimo: Decimal
    unidad: UnidadInventario
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class InventarioConProducto(InventarioOut):
    producto_nombre: str
    producto_disponible: bool


class InventarioUpdate(BaseModel):
    stock_minimo: Decimal | None = Field(
        default=None, ge=0, max_digits=12, decimal_places=3
    )
    unidad: UnidadInventario | None = None


class AjusteInventario(BaseModel):
    cantidad: Decimal = Field(gt=0, max_digits=12, decimal_places=3)
    tipo: TipoAjuste
    motivo: str | None = Field(default=None, max_length=255)

File: frontend/src/types/inventario.ts
export type UnidadInventario = "unidad" | "kg" | "g" | "l" | "ml";

export const UNIDADES: UnidadInventario[] = ["unidad", "kg", "g", "l", "ml"];

export type TipoAjuste = "entrada" | "salida";

export interface Inventario {
  id: number;
  id_producto: number;
  cantidad: string;
  stock_minimo: string;
  unidad: UnidadInventario;
  updated_at: string;
}

export interface InventarioConProducto extends Inventario {
  producto_nombre: string;
  producto_disponible: boolean;
}

export interface InventarioUpdate {
  stock_minimo?: number | string;
  unidad?: UnidadInventario;
}

export interface AjusteInventario {
  cantidad: number | string;
  tipo: TipoAjuste;
  motivo?: string;
}
